src/resolvers/exotics/git-resolver.js

```diff
@@ -142,7 +142,23 @@ export default class GitResolver extends ExoticResolver {
         return file;
       }
     }
-
-    throw new MessageError(this.reporter.lang('couldntFindManifestIn', transformedUrl));
+    // no registry files were found, in that case we fake a package.json thus allowing leaf dependencies without
+    // any registry file, same as bower
+    // TODO hack
+    const names = parts.path.split('/');
+    const registry = registries.npm;
+    const json = {
+      name: names[names.length - 1],
```

Comment:
That looks dangerous, what if `path` ends with `.git` suffix? The most logical way is to infer name from the key in `dependencies`:
```
"dependencies": {
  "desired-name": "some-repo-without-package-json"
}
```

Will be installed under `desired-name`.

As I understand that name is not available when request comes from CLI UI, right? That way maybe it's worth expanding CLI requests to `name=spec` syntax:

```
% yarn add desired-name=some-repo-without-package-json
```